osu.Game/Overlays/NewsOverlay.cs

```diff
@@ -56,13 +57,31 @@ private void load(OsuColour colours)
             };
 
             header.Current.BindTo(Current);
+
             Current.TriggerChange();
         }
 
-        public void ShowFrontPage()
+        private CancellationTokenSource loadChildCancellation;
+
+        protected void LoadChildContent(NewsContent newContent)
         {
-            Current.Value = null;
+            content.FadeTo(0.2f, 300, Easing.OutQuint);
+
+            loadChildCancellation?.Cancel();
+
+            LoadComponentAsync(newContent, c =>
+            {
+                content.Child = c;
+                content.FadeIn(300, Easing.OutQuint);
+            }, (loadChildCancellation = new CancellationTokenSource()).Token);
+        }
+
+        public void ShowArticle(string article)
+        {
+            Current.Value = article;
             Show();
         }
+
+        public void ShowFrontPage() => ShowArticle(null);
```

Comment:
This looks a bit like a "magic" API. What's this `article` string? An URL? Article slug? Why's `null` the front page? I'd like this to be easier to understand and read.

Reply:
Actually in the final implementation, the Current bindable should contain a struct about a specific news article but since the web API struct for news isn't defined yet, I use string as a placeholder for the moment. Asking to show article null should instead should show the news front page (whose implementation is to come later)